icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -94,7 +104,6 @@ const struct SIPrefixStrings {
 
 // TODO(ICU-21059): Get this list from data
 const char16_t* const gSimpleUnits[] = {
-    u"one", // note: expected to be index 0
```

Comment:
Since the SimpleUnitImpl default constructor defaults to index 0, I feel like we should keep an empty string here.

Reply:
I also changed SingleUnitImpl to use -1 - I'd have to change that back if we keep it here. My idea was to not confuse default-constructed and identifier-constructed SingleUnitImpl instances by treating them more differently.

Additionally I wasn't sure what the Trie would do with an empty string (UCharsTrieBuilder b; b.add(u"", ...);)

I'll leave this unresolved for now (first dealing with other feedback and sleeping), and either I'll come back to it after reviewing my decisions, or you'll come back to the decision while reviewing other changes related to it. :)

Reply:
Going the "-1" route has indeed created some pain in serializing. I'm far enough into this that I think I'll finish fixing it before I consider whether it's the wrong tree I'm barking up. :-/ If so, backtrack and restore the "dimensionless" SingleUnitImpl instances as "proper" instances.